diagnostics/forcing_feedback/forcing_feedback_kernelcalcs.py

```diff
@@ -0,0 +1,337 @@
+fluxanom_lw_q_tot_TOA[fluxanom_lw_q_tot_TOA==-float('Inf')] = np.nan
+fluxanom_lw_q_clr_TOA[fluxanom_lw_q_clr_TOA==-float('Inf')] = np.nan
+
+
+
+#Compute surface albedo change
+alb_pert_tot = model_mainvar_pert[os.environ["rsus_var"]]/model_mainvar_pert[os.environ["rsds_var"]]
+alb_climo_tot = model_mainvar_climo[os.environ["rsus_var"]]/model_mainvar_climo[os.environ["rsds_var"]]
+alb_pert_clr = model_mainvar_pert[os.environ["rsuscs_var"]]/model_mainvar_pert[os.environ["rsdscs_var"]]
+alb_climo_clr = model_mainvar_climo[os.environ["rsuscs_var"]]/model_mainvar_climo[os.environ["rsdscs_var"]]
+alb_pert_tot[np.isinf(alb_pert_tot)] = 0
+alb_climo_tot[np.isinf(alb_climo_tot)] = 0
+alb_pert_clr[np.isinf(alb_pert_clr)] = 0
+alb_climo_clr[np.isinf(alb_climo_clr)] = 0
+alb_pert_tot[alb_pert_tot>1] = 0
+alb_pert_tot[alb_pert_tot<0] = 0
+alb_climo_tot[alb_climo_tot>1] = 0
+alb_climo_tot[alb_climo_tot<0] = 0
+alb_pert_clr[alb_pert_clr>1] = 0
+alb_pert_clr[alb_pert_clr<0] = 0
+alb_climo_clr[alb_climo_clr>1] = 0
+alb_climo_clr[alb_climo_clr<0] = 0
+#Compute Surface Albedo Radiative Flux Anomalies
+[fluxanom_a_sfc_tot_TOA,fluxanom_a_sfc_clr_TOA] = \
+   fluxanom_calc_3D(alb_pert_tot,alb_climo_tot,sw_a_kern_TOA/.01,swclr_a_kern_TOA/.01,alb_pert_clr,alb_climo_clr)
+
+#Compute NET Radiative Flux Anomalies
+Rtot_LW_TOA_pert = -model_mainvar_pert[os.environ["rlut_var"]]
+Rtot_SW_TOA_pert = model_mainvar_pert[os.environ["rsdt_var"]] - model_mainvar_pert[os.environ["rsut_var"]]
+Rtot_LW_TOA_climo = -model_mainvar_climo[os.environ["rlut_var"]]
+Rtot_SW_TOA_climo = model_mainvar_climo[os.environ["rsdt_var"]] - model_mainvar_climo[os.environ["rsut_var"]]
+Rclr_LW_TOA_pert = -model_mainvar_pert[os.environ["rlutcs_var"]]
+Rclr_SW_TOA_pert = model_mainvar_pert[os.environ["rsdt_var"]]-model_mainvar_pert[os.environ["rsutcs_var"]]
+Rclr_LW_TOA_climo = -model_mainvar_climo[os.environ["rlutcs_var"]]
+Rclr_SW_TOA_climo = model_mainvar_climo[os.environ["rsdt_var"]]-model_mainvar_climo[os.environ["rsutcs_var"]]
+
+
+#TOA
+[fluxanom_Rtot_LW_TOA,fluxanom_Rclr_LW_TOA] = \
+      fluxanom_calc_3D(Rtot_LW_TOA_pert,Rtot_LW_TOA_climo,np.ones(lw_ts_kern_TOA.shape),np.ones(lwclr_ts_kern_TOA.shape), \
+                       Rclr_LW_TOA_pert,Rclr_LW_TOA_climo)
+[fluxanom_Rtot_SW_TOA,fluxanom_Rclr_SW_TOA] = \
+   fluxanom_calc_3D(Rtot_SW_TOA_pert,Rtot_SW_TOA_climo,np.ones(lw_ts_kern_TOA.shape),np.ones(lwclr_ts_kern_TOA.shape), \
+                       Rclr_SW_TOA_pert,Rclr_SW_TOA_climo)
+#TOA CRE
+fluxanom_Rcre_LW_TOA = fluxanom_Rtot_LW_TOA - fluxanom_Rclr_LW_TOA
+fluxanom_Rcre_SW_TOA = fluxanom_Rtot_SW_TOA - fluxanom_Rclr_SW_TOA
+
+
+#Compute Instantaneous Radiative Forcing as difference between NET Radiative Flux Anomalies and
+#the sum of all indivisual radiative flux anomalies. Total-sky IRF computed as 
+#Clear-Sky IRF divided by cloud masking constant. NOTE, these cloud masking constants may not apply
+#to user's specific model experiment.
```

Comment:
If the cloud masking constants ("1.24" and "2.43" below) are nontrivial, it may be desirable to expose them to the user as environment variables in the settings file, rather than hard-coding them here.